### DISCUSSION TITLE
How to use your own Class images?

### DISCUSSION THREAD

**Discussion by nanafy** [d8ahazard/sd_dreambooth_extension#570] [Q&A]
I know it sounds like a dumb question. much help would be appreciated. I knew how to use the older version of this extension like a champ. All these new settings have been confusing for me to adapt to. I am just wondering, what steps do I have to take to ensure my class images that I pre-generated are being considered?

Steps I take

-Uncheck Generate Concept Images Using txt2img
-Give directories a path to my images
-In image generation tab, set total number of class/reg images to 0 images

I think I am doing something wrong, as in the previous versions, I would see the images being passed into the cache. It would first cache my class images and then start training. But now I don't see that step.

**Comment by zahdab** [comment]
you specify the folder where ur class images are under concepts : Classification Dataset Directory, set the generate class images to 0

**Reply by nanafy** [reply]
Thanks for your quick reply. Just wanted to clarify, are you talking about 1 or 2
![image](https://user-images.githubusercontent.com/19253644/208269426-287d818d-e96d-480b-a28d-3c88019378fd.png)

**Reply by nanafy** [reply]
I believe you meant 1, I have been doing that but I never see them being cached in cmd when I start training. It used to say caching  0/800 and then it would start training. Hence my assumption of incorrectly setting it up.

**Comment by nanafy** [comment]
@d8ahazard could you perhaps help me get to the bottom of this one. I know you are busy, much help would be appreciated, when you have some time.  Just to re-iterate, I used to see my class images being cached in the cmd window before the training would begin. If I had 800 images, it would say caching latents 0/800. Now it just starts the training. I am going on a whim here and thinking that my class images are not being utilized from the specified directory.

**Reply by cohan8999** [reply]
Have you remembered to set "Total Number of Class/Reg Images" to the same amount of class images you have? It should not be 0 as is default

**Reply by nanafy** [reply]
Yes I posted the image above of my settings. I have it to set to a non-zero value.

**Reply by nanafy** [reply]
I put number of samples to generate as 0 because I wanted to use my own. But I just don't know if its doing anything as I don't see them being cached at the start like the older versions

**Reply by cohan8999** [reply]
That's weird. It works for me. Maybe try doing a fresh install of A1111?

**Reply by nanafy** [reply]
How do you know, can you see in cmd that its caching latents? 
I will do a fresh install tonight. Let me know what I have to look out for!

**Comment by alicedelish** [comment]
You just specify your classifier images folder in the appropriate field, and set "Total number of class\reg images" to the amount of pics you have in your folder. If the number you enter is more than that - dreambooth will generate missing picture.
